doc/architecture/software-update.rst

```diff
@@ -17,53 +17,128 @@ Ostro OS based software can be deployed to a target device in two ways:
 
 
 Software Update Client
 ======================
 
-The client runs on the device and is responsible of downloading, verifying
+The client runs on the device and is responsible for downloading, verifying
 and applying the updates.
-It can also restore modified files to their pristine state as then were in the
-initial sw image.
+It can also restore modified files to their pristine state as they were in the
+initial SW image.
+
+.. note::
+   The version of swupd client that ships with Ostro 1.0 does not support yet
+   signature verification, therefore it is necessary to provide the swupd feed
+   through a secure channel (https).
+   Once this is addressed, also unsecure channels will be suitable, because,
+   as long as the signature and integrity checks are satisfied,
+   the software update client will treat them as trusted.
 
 
 SW Bundles
 ==========
 
 The software update mechanism supports the concept of bundles_: groups of files
 that provide one functionality.
-It is similar to the concept of a package used in the vast majority of Linux distros,
-but in general one software update bundle maps to several packages (ex: .rpm or .deb)
 
-There is one main os-core bundle containing everything that is mandatory for having
-a bare-bones working system.
-Everything else is optional and can be either deployed or removed by using the
+Ostro OS users coming from popular Linux distros are probably already familiar
+with the concept of package.
+How is a bundle different from a package?
+
+- A bundle has no dependency: it is self-contained.
+  A bundle might also list files contained in other bundles, and that is
+  permitted by design. The SW Update Server and client work together to handle
+  such overlapping.
+
+- As long as bundles come from the same SW Update stream, they cannot cause
+  installation conflicts on the device, because they were obtained from the same
+  rootfs.
+  It is expected that conflicts were solved during the creation of the rootfs.
+  The team managing the SW Update stream is responsible for vetting the rootfs used
+  during the creation of bundles.
+
+- Bundle creation implies that there is a starting rootfs that provides all the
+  files the bundles will include. How this rootfs is created is irrelevant from
+  the perspective of the SW Update Server tool creating the bundles.
+  In Ostro OS, the rootfs is populated through the execution of bitbake recipes,
+  but in Clear Linux this is done by installing .rpm packages.
+  Compared to packages, bundles are a competing way to deliver files to the
+  target device, however bundles can be inserted in an existing workflow that is
+  based on packages, with minimal modifications.
+
+Coming back to how bundles relate to each other: there is one special bundle,
+called "os-core", which contains the minimum set of files needed to have a
+barebone working system.
+
+All other bundles are optional and can be either deployed or removed by using the
```

Comment:
On 26 April 2016 at 14:34, Geoffroy Van Cutsem <notifications@github.com>
wrote:

>  +All other bundles are optional and can be either deployed or removed by
> using the
>
> Can we actually already remove bundles or is it something that will be
> implemented later?
>

​It works, I confirmed it on the image​

​I cre​ated with the preinstalled bundle and I was able to remove it.

The fact it requires to lookup the remote manifest might not be acceptable
in certain corner cases, but overall I think it's ok.


-- 
igor